src/content/docs/apm/agents/net-agent/net-agent-api/net-agent-api.mdx

```diff
@@ -0,0 +1,2579 @@
+    >
+    
+    Provides access to the currently executing [span](/docs/apm/agents/net-agent/net-agent-api/ispan/), making [span-specific methods](/docs/apm/agents/net-agent/net-agent-api/ispan) available within the New Relic API.
+
+    ### Example
+
+    ```cs
+    IAgent agent = NewRelic.Api.Agent.NewRelic.GetAgent(); 
+    ITransaction transaction = agent.CurrentTransaction; 
+    ISpan currentSpan = transaction.CurrentSpan;
+    ```
+
+    </Collapser>
+    <Collapser
+        id="SetUserId"
+        title="SetUserId"
+    >
+    Associates a User Id with the current transaction.
+
+    This method requires .NET agent and .NET agent API [version 10.9.0](/docs/release-notes/agent-release-notes/net-release-notes/net-agent-1090) or higher.
+
+    ### Syntax
+
+    ```cs
+    ITransaction SetUserId(string userId)
+    ```
+
+    ### Parameters
+
+    <table>
+    <thead>
+        <tr>
+        <th>
+            Parameter
+        </th>
+
+        <th>
+            Description
+        </th>
+        </tr>
+    </thead>
+
+    <tbody>
+        <tr>
+        <td>
+            `userId`
+
+            _string_
+        </td>
+
+        <td>
+            The User Id to be associated with this transaction.
+
+            * `null`, empty and whitespace values will be ignored.
+        </td>
+        </tr>
+
+    </tbody>
+    </table>
+
+    ### Example
+
+    ```cs
+    IAgent agent = NewRelic.Api.Agent.NewRelic.GetAgent(); 
+    ITransaction transaction = agent.CurrentTransaction; 
+    transaction.SetUserId("BobSmith123");
+    ```
+    </Collapser>
+</CollapserGroup>
+    </Collapser>
+    <Collapser
+        id="ISpan"
+        title="ISpan"
+    >
+    ### Syntax
+
+    ```cs
+    Public interfac ISpan
+    ```
+
+    Provides access to span-specific methods in the New Relic API.
+
+    ### Description
+
+    Provides access to span-specific methods in the New Relic .NET agent API. To obtain a reference to `ISpan`, use:
+
+    * The `CurrentSpan` property on [`IAgent`](/docs/agents/net-agent/net-agent-api/iagent) (Recommended).
+    * The `CurrentSpan` property on [`ITransaction`](/docs/agents/net-agent/net-agent-api/itransaction).
+
+    This section contains descriptions and parameters of `ISpan` methods:
+
+    <table>
+    <thead>
+        <tr>
+        <th>
+            Name
+        </th>
+
+        <th>
+            Description
+        </th>
+        </tr>
+    </thead>
+
+    <tbody>
+        <tr>
+        <td>
+            [`AddCustomAttribute`](#addcustomattribute)
+        </td>
+
+        <td>
+            Add contextual information from your application to the current span in form of attributes.
+        </td>
+        </tr>
+        <tr>
+        <td>
+            [`SetName`](#setname)
+        </td>
+
+        <td>
+            Changes the name of the current span/segment/metrics that will be reported to New Relic.
+        </td>
+        </tr>
+    </tbody>
+    </table>
+
+    ### AddCustomAttribute [#addcustomattribute]
```

Comment:
This seems to be missing a collapser to make it its own entry
![Uploading image.png…]()